### ISSUE TITLE
API error shadowed by ClassCastException message if error is not a json object

### ISSUE THREAD

**Issue by notfinch** [stripe/stripe-java#1846]
### Describe the bug

Using some stripebased payments, when an unsuccessful response is resived, the "error" field in the body may be a string instead of a correct json object in raw response
While parsing the error in the com.stripe.net.LiveStripeResponseGetter#handleApiError method, a ClassCastException error is thrown which is thrown out and is uninformative to the sdk consumer

### To Reproduce

1. Make any api call that returns a non json type error
2. Get exception `java.lang.ClassCastException: class com.google.gson.JsonPrimitive cannot be cast to class com.google.gson.JsonObject`

### Expected behavior

It is expected that any api call with an error will throw a StripeException or its child exception with a message from the api

### OS

all

### Java version

21

### stripe-java version

26.1.0

### API version

2024-06-20

### Additional context

```
try {
      JsonObject jsonObject =
          ApiResource.INTERNAL_GSON
              .fromJson(response.body(), JsonObject.class)
              .getAsJsonObject("error");
      error = ApiResource.deserializeStripeObject(jsonObject.toString(), StripeError.class, this);
    } catch (JsonSyntaxException e) {
      raiseMalformedJsonError(response.body(), response.code(), response.requestId(), e);
    }
```
The issue here `getAsJsonObject("error")`
I think that just need to add ClassCastException to the catch block

**Comment by remi-stripe** [comment]
@notfinch it should not be possible for the error to be a string instead of a proper JSON object. If this happens this would be a bug on our end we'd need to fix. I have definitely seen us do this incorrectly in the past but we've always quickly fixed those when it happened.

Do you have a concrete example where this happened and what the raw response looked like?

**Comment by omkarkulkarni2704** [comment]
To fix the issue, add a ClassCastException catch block to handle cases where the error field is a string instead of a JsonObject and throw an informative StripeException.

**Comment by xavdid-stripe** [comment]
@omkarkulkarni2704 were you seeing this issue as well? Do you (or anyone else) have a request ID to share so we can take a closer look?

**Comment by omkarkulkarni2704** [comment]
@xavdid-stripe Yes, I encountered this issue but I don’t have a specific```request_id``` to share. The problem occurs when the ```error``` field is returned as a string instead of a JSON object, leading to a ```ClassCastException```. 
Let me know how I can assist further!

**Comment by xavdid-stripe** [comment]
Right, but that really shouldn't happen.

If you notice it again, please note the request id and pass it along here! that'll help us figure out what's up

**Comment by omkarkulkarni2704** [comment]
Ok

**Comment by riken127** [comment]
Hi! I checked the current implementation before proposing a change.

`parseStripeError` still calls `getAsJsonObject("error")` directly and only catches `JsonSyntaxException`. However, `handleError` now separately detects string primitives and routes them through the OAuth error path.

The remaining problematic cases appear to be malformed API responses where `error` is an array, boolean, number, null, or otherwise not a JSON object.

Would you accept a defensive parsing change that explicitly validates the element type and routes invalid shapes through `makeMalformedJsonError`, instead of allowing a Gson type exception to escape?

I would include focused tests for those malformed shapes while preserving the existing OAuth string handling.

**Comment by jar-stripe** [comment]
Hi @riken127 !  thanks for the offer!  Have you been able to reproduce this issue?  We would really rather understand the issue and fix the API if it is misbehaving here.
